### PR TITLE
Bump Microsoft.OWin to 4.0 and Microsoft.AspNetCore.Mvc.Core 2.0.1

### PR DESCRIPTION
**Pull request by xuzhg**
### Issues

*This pull request fixes #xxx.*

### Description

*Briefly describe the changes of this pull request.*

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/Microsoft.Test.E2E.AspNet.OData.csproj [hunk 1]
```diff
@@ -47,8 +47,8 @@
       <HintPath>..\..\..\..\sln\packages\Microsoft.OData.Edm.7.12.1\lib\net45\Microsoft.OData.Edm.dll</HintPath>
       <Private>True</Private>
     </Reference>
-    <Reference Include="Microsoft.Owin, Version=2.0.2.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <HintPath>..\..\..\..\sln\packages\Microsoft.Owin.2.0.2\lib\net45\Microsoft.Owin.dll</HintPath>
+    <Reference Include="Microsoft.Owin, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
+      <HintPath>..\..\..\..\sln\packages\Microsoft.Owin.4.0.0\lib\net451\Microsoft.Owin.dll</HintPath>
```
Comment (ElizabethOkerio) [inline test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/Microsoft.Test.E2E.AspNet.OData.csproj:51]: `net452`? I see the packagereference has a `net452 targetframework`

**Review by ElizabethOkerio — APPROVED** [review]

**Review by gathogojr — DISMISSED** [review]

**Review by KenitoInc — DISMISSED** [review]

**Comment by robertmclaws** [comment]
Question on this: if .NET Core 2.0 is no longer supported by Microsoft, why not bump the version to 3.1.0? That's the earliest supported version now, and even that support ends before the end of the year.

**Comment by robertmclaws** [comment]
Along the same vein, .NET 4.6.2 is now the earliest supported .NET Framework runtime. https://docs.microsoft.com/en-us/lifecycle/products/microsoft-net-framework

**Comment by xuzhg** [comment]
> Question on this: if .NET Core 2.0 is no longer supported by Microsoft, why not bump the version to 3.1.0? That's the earliest supported version now, and even that support ends before the end of the year.

This PR is simply fixing the security issue for dependencies. What you said is a great topic that we have taken into consideration. Thanks.

**Pull request merged**